<a href="https://colab.research.google.com/github/anabiarochar/Mestrado_AlgoritmosProgramacao/blob/main/Projeto1/1_Bolha2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_1000_aleatorio.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 0.050899 segundos
[1, 3, 8, 9, 9, 10, 10, 10, 10, 12]


###Aleatório 1000:

Tentativa 1 - Tempo de execução: 0.083225 segundos   
Tentativa 2 - Tempo de execução: 0.051019 segundos  
Tentativa 3 - Tempo de execução: 0.050137 segundos  
Tentativa 4 - Tempo de execução: 0.050899 segundos

In [8]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_10000_aleatorio.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 7.299729 segundos
[3, 3, 4, 8, 9, 9, 9, 11, 12, 12]


###Aleatório 10000:

Tentativa 1 - Tempo de execução: 6.349090 segundos   
Tentativa 2 - Tempo de execução: 6.380606 segundos  
Tentativa 3 - Tempo de execução: 7.478226 segundos  
Tentativa 4 - Tempo de execução: 7.299729 segundos

In [9]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_100000_aleatorio.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 776.809246 segundos
[2, 5, 6, 6, 8, 9, 10, 13, 14, 14]


###Aleatório 100000:

Tentativa 1 - Tempo de execução: 776.446981 segundos  
Tentativa 2 - Tempo de execução: 765.963112 segundos  
Tentativa 3 - Tempo de execução: 735.920505 segundos  
Tentativa 4 - Tempo de execução: 776.809246  segundos

In [14]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_1000_crescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 0.029674 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Crescente 1000:

Tentativa 1 - Tempo de execução: 0.032423 segundos   
Tentativa 2 - Tempo de execução: 0.029141 segundos  
Tentativa 3 - Tempo de execução: 0.053785 segundos  
Tentativa 4 - Tempo de execução: 0.029674 segundos

In [18]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_10000_crescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 3.564551 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Crescente 10000:

Tentativa 1 - Tempo de execução: 3.682419 segundos   
Tentativa 2 - Tempo de execução: 3.585820 segundos  
Tentativa 3 - Tempo de execução: 3.565169 segundos  
Tentativa 4 - Tempo de execução: 3.564551 segundos

In [20]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_100000_crescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 414.604800 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Crescente 100000:

Tentativa 1 - Tempo de execução: 417.514118 segundos   
Tentativa 2 - Tempo de execução: 414.604800 segundos  
Tentativa 3 - Tempo de execução: 415.233021 segundos  
Tentativa 4 - Tempo de execução: 419.112301 segundos

In [24]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_1000_decrescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 0.062281 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Decrescente 1000:

Tentativa 1 - Tempo de execução: 0.062020 segundos   
Tentativa 2 - Tempo de execução: 0.063244 segundos  
Tentativa 3 - Tempo de execução: 0.062244 segundos  
Tentativa 4 - Tempo de execução: 0.062281 segundos

In [28]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_10000_decrescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 8.880975 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Decrescente 10000:

Tentativa 1 - Tempo de execução: 8.906069 segundos   
Tentativa 2 - Tempo de execução: 7.438931 segundos  
Tentativa 3 - Tempo de execução: 8.888610 segundos  
Tentativa 4 - Tempo de execução: 8.880975 segundos

In [29]:
import requests
import time

# Importar dados
url = "https://raw.githubusercontent.com/anabiarochar/Mestrado_AlgoritmosProgramacao/main/Projeto1/Vetores/vetor_100000_decrescente.txt"

try:
    response = requests.get(url)
    content = response.text

    # Ler arquivo como vetor separado por vírgula
    alist = [int(x.strip()) for x in content.split(',') if x.strip()]

except Exception as e:
    print(f"Erro ao carregar ou processar o arquivo: {e}")

# Método Bolha
def bubbleSort(alist):
  for passnum in range(len(alist)-1,0,-1):
    for i in range(passnum):
      if alist[i]>alist[i+1]:
        temp = alist[i]
        alist[i] = alist[i+1]
        alist[i+1] = temp


# Execução e Tempo
if alist:
    inicio = time.perf_counter()
    vetor_ordenado = bubbleSort(alist)
    fim = time.perf_counter()

    tempo_total = fim - inicio

    print(f"Tempo de execução: {tempo_total:.6f} segundos")
    print(alist[:10])

else:
    print("Vetor vazio")

Tempo de execução: 863.781448 segundos
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


###Decrescente 100000:

Tentativa 1 - Tempo de execução: 858.927049 segundos  
Tentativa 2 - Tempo de execução: 858.957241 segundos  
Tentativa 3 - Tempo de execução: 855.807456 segundos  
Tentativa 4 - Tempo de execução: 863.781448 segundos  

In [10]:
!echo "--- CPU Info ---"
!lscpu | grep "Model name"
!echo "--- RAM Info ---"
!free -h

--- CPU Info ---
Model name:                              Intel(R) Xeon(R) CPU @ 2.20GHz
--- RAM Info ---
               total        used        free      shared  buff/cache   available
Mem:            12Gi       818Mi       8.7Gi       1.0Mi       3.2Gi        11Gi
Swap:             0B          0B          0B
